src/parser/Parser.ts

```diff
@@ -2839,6 +2985,19 @@ export class References {
     public namespaceStatements = [] as NamespaceStatement[];
     public newExpressions = [] as NewExpression[];
     public propertyHints = {} as Record<string, string>;
+
+    public getContainingClass(currentToken: Token): ClassStatement {
```

Comment:
These don't really seem to belong on the `References` class. Instead, we should add these as methods to the `Parser` class.